ajax/ajaxUndoVote.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+include('config/init.php');
+
+if(!isset($_SESSION['userID'])) {  //&& !empty($_SESSION['userID']) <-- had this before. not sure if necessary
+	echoNotLoggedIn("You must be logged in to vote.");
+	die();
+}
+
+$eventID=$_REQUEST['eventID'];
+$sessionUserID=$_SESSION['userID'];
+$userVote = doesUserVoteExist($eventID, $sessionUserID);
+
+deleteUserVote($userVote['0']['eventID'], $_SESSION['userID']);
+
+
+// the following wasn't working :( but not sure why because I thought it was similiar logic to the front pages. 
+
+// if(($_REQUEST['ifUndoAndVote']!=null && $_REQUEST['direction'] == "up") || ($_REQUEST['ifUndoAndVote']==null && $_REQUEST['direction'] == "down")){
+// 	generalVoteInDB($eventID, "up");
+// }
+// else{
+// 	generalVoteInDB($eventID, "down");
+// }
+
+
+if($_REQUEST['ifUndoAndVote']!=null){ //if you have to undo the vote and then vote
+	if($_REQUEST['direction'] == "up"){
+		upVoteInDB($eventID);
+	}
+	else{ // aka direction is down
+		downVoteInDB($eventID);
+	}
+}
+else{	//if this is just undo-ing the vote, no next steps
+	if($_REQUEST['direction'] == "up"){
```

Comment:
This is minor, but I'd probably make a `voteInDB` function that takes a direction. That way calling it will be cleaner, and you won't need this if statement here (it'll go in the function instead, and it's normally better to hide complexity in functions so this code is cleaner)

Reply:
Also, since this is the undo endpoint, I don't think you need to handle voting here if you take my other advice,